boa_engine/src/builtins/function/mod.rs

```diff
@@ -474,7 +475,9 @@ impl BuiltInFunctionObject {
                     parameters
                 };
 
-            let body_arg = body_arg.to_string(context)?;
+            // TODO: `Parser` should probably take `u32` unicode codepoints
+            // to avoid converting the whole string before parsing
```

Comment:
No, because the parser wouldn't be capable of parsing `UTF-8` without converting it to `UTF-16` first.
Also, we would have to convert from `UTF-16` to Unicode codepoints each time the lexer has to take the next `char`, so it's easier to ask for an `u32` iterator in the first place.